docs/developer/solutions/archive.rst

```diff
@@ -0,0 +1,177 @@
+Solution archive guidelines
+===========================
+
+To provide a predictable interface with packaged Solutions, MetalK8s expects a
+few criteria to be respected, described below.
+
+
+Archive format
+--------------
+
+Solution archives must use the `ISO-9660:1988`_ format, including `Rock Ridge`_
+and Joliet_ directory records. The character encoding must be UTF-8_. The
+conformance level is expected to be at most 3, meaning:
+
+- Directory identifiers may not exceed 31 characters (bytes) in length
+- File name + ``'.'`` + file name extension may not exceed 30 characters
+  (bytes) in length
+- Files are allowed to consist of multiple sections
+
+The generated archive should specify a volume ID, set to
+``{project_name} {version}``.
+
+.. todo::
+
+   Clarify whether Joliet/Rock Ridge records supersede the conformance level
+   w.r.t. filename lengths
+
+
+.. _`ISO-9660:1988`: https://www.iso.org/obp/ui/#iso:std:iso:9660:ed-1:v1:en
+.. _`Rock Ridge`: https://en.wikipedia.org/wiki/Rock_Ridge
+.. _Joliet: https://en.wikipedia.org/wiki/Joliet_(file_system)
+.. _UTF-8: https://tools.ietf.org/html/rfc3629
+
+Here is an example invocation of the common Unix mkisofs_ tool to generate such
+archive::
+
+   mkisofs
```

Comment:
I was thinking, for this (and how to generate images with `skopeo`/`hardlink`), would it be worth to extract those "functions" of our build system into some sort of tools/library/whatever that then could be used by both our buildchain AND the solutions?

That would avoid the "reverse-engineer what our buildchain does, put it in the doc and take care of keeping both of them in sync" approach.

Reply:
Yes, providing shared tooling would be great. I think we'll know more what are the requirements once we start working on a minimal Solution implementation (#1060).